### PR TITLE
Remove the usage of TabRepository in PrivacyPracticesActivity

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/0/1201307807827049/f

### Description
Removed the reference to tab repository from PrivacyPracticesActivity.

### Steps to test this PR

- install / upgrade from this branch
- navigate to a website (e.g https://www.amazon.com, https://www.theguardian.com)
- tap on the score bubbles displayed at the left side of the URL and you'll open the "Privacy Dashboard" screen
- on "Privacy Dashboard" screen tap on "Privacy Practices" (can be "Poor Privacy Practices" / "Unknown Privacy Practices" / etc) and you'll open the "Privacy Practices" screen
-  check the "Privacy Practices" screen behaves as expected (correct information displayed for poor / unknown / etc)

### Video

https://user-images.githubusercontent.com/7963079/140561826-a635f132-bded-4c51-8430-0ca97f9b568e.mp4

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyPracticesActivity.kt [hunk 1]
```diff
@@ -62,16 +55,11 @@ class PrivacyPracticesActivity : DuckDuckGoActivity() {
...
-
-        repository.retrieveSiteData(intent.tabId!!).observe(
-            this,
-            Observer<Site> {
-                viewModel.onSiteChanged(it)
-            }
-        )
+        lifecycleScope.launch {
+            viewModel.privacyPractices(intent.tabId!!)
+                .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyPracticesActivity.kt:60]: Changed the minimum active state to STARTED.

File: app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyPracticesViewModel.kt [hunk 2]
```diff
@@ -39,42 +41,28 @@ class PrivacyPracticesViewModel : ViewModel() {
         val badTerms: List<String> = emptyList()
     )
 
-    private val viewState = MutableStateFlow(ViewState())
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyPracticesViewModel.kt:42]: Removed this and the viewState() function as the initial state is provided with the usage of "stateIn".

Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/privacy/ui/PrivacyPracticesViewModel.kt:42]: Nice!

**Review by joshliebe — APPROVED** [review]
Works as expected ✅ Good job!

**Comment by anikiki** [comment]
> Works as expected white_check_mark Good job!

Thanks for the review @joshliebe :tada:

**Pull request merged**